File: tests/models/Order.ts
import { z } from 'zod';
import { OrderItemSchema } from './OrderItem';

export const OrderSchema = z.object({
  id: z.number(),
  customerId: z.number(),
  orderDate: z.string(),
  totalAmount: z.number(),
  status: z.string(),
  shippingAddressId: z.number(),
  items: z.array(OrderItemSchema),
});

export type Order = z.infer<typeof OrderSchema>;

File: api/OnlineStore.Api/Models/Order.cs
namespace OnlineStore.Api.Models;

public class Order
{
    public int Id { get; set; }
    public int CustomerId { get; set; }
    public DateTime OrderDate { get; set; }
    public decimal TotalAmount { get; set; }
    public string Status { get; set; } = string.Empty;
    public int ShippingAddressId { get; set; }
    public List<OrderItem> Items { get; set; } = new();
}
